src/simulator/simulator.cpp

```diff
@@ -364,76 +363,25 @@ void Simulator::publishVisualizationMarkers() {
   objectLinesPublisher.publish(objectLinesMarker);
 }
 
-float AbsBound(float x, float bound) {
-  if (x > 0.0 && x > bound) {
-    return bound;
-  } else if (x < 0.0 && x < -bound) {
-    return -bound;
-  }
-  return x;
-}
-
-void Simulator::DriveCallback(const AckermannCurvatureDriveMsg& msg) {
- if (!isfinite(msg.velocity) || !isfinite(msg.curvature)) {
-    printf("Ignoring non-finite drive values: %f %f\n",
-           msg.velocity,
-           msg.curvature);
-    return;
-  }
-  last_cmd_ = msg;
-  tLastCmd = GetMonotonicTime();
-}
-
 void Simulator::update() {
-  static const double kMaxCommandAge = 0.1;
-  if (GetMonotonicTime() > tLastCmd + kMaxCommandAge) {
-    last_cmd_.velocity = 0;
-  }
-  // Epsilon curvature corresponding to a very large radius of turning.
-  static const float kEpsilonCurvature = 1.0 / 1E3;
-  // Commanded speed bounded to motion limit.
-  const float desired_vel = AbsBound(last_cmd_.velocity, cMaxSpeed);
-  // Maximum magnitude of curvature according to turning limits.
-  const float max_curvature = 1.0 / cMinTurnR;
-  // Commanded curvature bounded to turning limit.
-  const float desired_curvature = AbsBound(last_cmd_.curvature, max_curvature);
-  // Indicates if the command is for linear motion.
-  const bool linear_motion = (fabs(desired_curvature) < kEpsilonCurvature);
-
-  const float dv_max = cDT * cMaxAccel;
-  const float bounded_dv = AbsBound(desired_vel - vel, dv_max);
-  vel = vel + bounded_dv;
-  const float dist = vel * cDT;
-
-  float dx = 0, dy = 0, dtheta = 0;
-  if (linear_motion) {
-    dx = dist;
-    dy = 0;
-    dtheta = cDT * cAngularErrorBias;
-  } else {
-    const float r = 1.0 / desired_curvature;
-    dtheta = dist * desired_curvature +
-        angular_error_(rng_) * cDT * cAngularErrorBias +
-        angular_error_(rng_) * cAngularErrorRate * fabs(dist *
-            desired_curvature);
-    dx = r * sin(dtheta);
-    dy = r * (1.0 - cos(dtheta));
-  }
-
-  curLoc.x() += dx * cos(curAngle) - dy * sin(curAngle);
-  curLoc.y() += dx * sin(curAngle) + dy * cos(curAngle);
-  curAngle = AngleMod(curAngle + dtheta);
+  // Step the motion model forward one time step
+  motion_model_->Step(cDT);
```

Comment:
I know this is irrelevant to current issue, but I think we would want a vector of EntityBase pointers to represent all robot motion models we need in the simulator? So the update would be similar to the update that is done for multiple obstacles.

Reply:
Are you suggesting a vector in order to handle the multi-robot case, or a vector because we will have different robot types? I'm not currently handling the multi-robot issue at all. I want to leave those changes for later. 

I have another plan for handling selecting the correct motion model type.

Reply:
I was suggesting a vector to handle multi-robot case. I guess it is not needed now, but we probably need to put more thoughts on how to deal with that

Reply:
Right this is an issue to consider at the next stage, after getting our models in and testing them.